### PR TITLE
fix(types): avoid any in react.ts

### PR DESCRIPTION
**Pull request by dai-shi**
There was a `TODO` in react.ts.
I originally thought this requires StoreMutator, but it doesn't actually mutate. So, it's rather simple?

### CODE REVIEW

File: src/react.ts [hunk 1]
```diff
@@ -10,32 +10,39 @@ import createStore, {
...
 export function useStore<TState extends State, StateSlice>(
-  api: StoreApi<TState>,
+  api: WithReact<StoreApi<TState>>,
   selector: StateSelector<TState, StateSlice> = api.getState as any,
   equalityFn?: EqualityChecker<StateSlice>
 ) {
   const slice = useSyncExternalStoreWithSelector(
     api.subscribe,
     api.getState,
-    // TODO avoid `any` and add type only in react.ts
```
Comment (dai-shi) [inline src/react.ts:27]: We simply allow specifying optional `getServerState` property. Note this has nothing to do with vanilla.ts.

**Review by devanshj — APPROVED** [review]
Looks good.

> I originally thought this requires StoreMutator, but it doesn't actually mutate. So, it's rather simple?

Yep, only middlewares require `StoreMutator`.

File: src/react.ts (hunk 1 above)
Comment (Andarist) [reply src/react.ts:27]: Out of curiosity - do you have any examples of how Zustand consumers should utilize `getServerState` nowadays? When I think about it (in the context of just any state management library) it seems quite cumbersome to provide this, especially if a given library supports multiple stores. I wonder if we could figure out a standard way for approaching this problem.

Comment (dai-shi) [reply src/react.ts:27]: > Out of curiosity - do you have any examples of how Zustand consumers should utilize `getServerState` nowadays?

No. Here, the API is just exposed, hoping to find a way to utilize it somehow.

> I wonder if we could figure out a standard way for approaching this problem.

Definitely. In the zustand case, I was hoping if someone come up with some ideas or concrete use cases. There might be a standard way.

**Pull request merged**